x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/date_range_picker.tsx

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useDispatch } from 'react-redux';
+import React, { memo, useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import moment, { Moment } from 'moment';
+import { EuiFlexGroup, EuiFlexItem, EuiDatePicker, EuiDatePickerRange } from '@elastic/eui';
+
+import * as i18 from '../../translations';
+import { useEndpointSelector } from '../../hooks';
+import { getActivityLogDataPaging, selectedAgent } from '../../../store/selectors';
+
+const DatePickerWrapper = styled.div`
+  width: ${(props) => props.theme.eui.fractions.single.percentage};
+  background: white;
+`;
+const StickyFlexItem = styled(EuiFlexItem)`
+  position: sticky;
+  top: ${(props) => props.theme.eui.euiSizeM};
+  z-index: 1;
+`;
+
+export const DateRangePicker = memo(() => {
+  // clear dates on endpoint selection
+  const elasticAgentId = useEndpointSelector(selectedAgent);
+  useEffect(() => {
+    return () => {
+      if (elasticAgentId) {
+        setStartDate(undefined);
+        setEndDate(undefined);
+      }
+    };
+  }, [elasticAgentId]);
+
+  const dispatch = useDispatch();
+  const {
+    page,
+    pageSize,
+    startDate: initialStartDate,
+    endDate: initialEndDate,
+  } = useEndpointSelector(getActivityLogDataPaging);
+  const [startDate, setStartDate] = useState<Moment | undefined>(
+    initialStartDate ? moment(initialStartDate) : undefined
+  );
+  const [endDate, setEndDate] = useState<Moment | undefined>(
+    initialEndDate ? moment(initialEndDate) : undefined
+  );
+
+  const onChangeStartDate = useCallback(
+    (date) => {
+      setStartDate(date);
```

Comment:
Refactor proposal: As we already have this keys in the redux store, we could use those instead of this local ones.
So, you can read the endDate and startDate from redux with a selector and you can dispatch an action to change it. 
Each time an action to change those dates is dispatched you can check if is not undefined then dispatch the search in the middleware so no need to setLocalState and then dispatchReduxAction. 
Let me know what do you think about it 🙂

Reply:
Nice. 👍  This also would solve the issue of clearing the dates on endpoint selection. ☝️  Although, I've to dispatch a new action to clear the dates when the date range is invalid.

Reply:
done f551b67d661a330621fb77c435db26ec90288b81